Run time system

ShayParsa opened the pull request.

ShayParsa committed:
Create RunTime

ShayParsa committed:
Delete RunTime

ShayParsa committed:
Create RunTime

ShayParsa committed:
Delete RunTime

ShayParsa committed:
Create RunTime

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class RunTime {
```
can you please rename the class to "RuntimeSystem" to stay consistant with the documentation

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+
+public class RunTime {
+
+	public ArrayList<String> listOfLines(String str) {
```
These helper methods should be made private

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+
+	public ArrayList<String> listOfLines(String str) {
+
+		ArrayList<String> lines = new ArrayList<>();
```
While ArrayList is good you should store your ArrayList object in a List object which will make the code more flexible (if you need help doing that I can show you how)

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+
+	public static boolean isLengthGreaterThanZero(String str) {
+
+		return str.length() > 0 && str != null;
```
you should be checking for null first and then for the length because right now the code might throw a null pointer exception

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+		return str.length() > 0 && str != null;
+	}
+
+	private boolean isSyntaxCorrect(String str) {
```
you don't need this function as the Parser will handle that so you can just directly call it

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+	}
+
+	public String splitLine(String str) {
+
```
I'm not sure I understand what this method is meant to do

Aether700 commented on src/NAR/RunTime:
```diff
@@ -0,0 +1,77 @@
+
+				if (!isSyntaxCorrect(statement)) {
+
+					System.out.print("!!! syntax error");
```
The print statements are fine for testing but we will need to remove them

Aether700 reviewed (commented):
The RuntimeSystem class should be a singleton which it is not. If you need help implementing the singleton pattern or if you don't know what a singleton is let me know and I can help you with that and/or explain it

ShayParsa committed:
Update RunTime

about the method reduceSpace : The input would be one line of the input String. First, it reduces all consecutive spaces to only one space. Then check if there is still a white space at the first Char to delete it. To check the first char value, we first need to make sure if the length of that line/string is greater than zero. I changed the method name from "splitLine" to reduceSpace to make more sense.

Aether700 commented on src/NAR/RunTime:
```diff
@@ -40,35 +45,27 @@ public class RunTime {
 
-					System.out.print("!!! syntax error");
+				System.out.print(statement);
+                //check if the syntax is correct
```
the formatting of the tabs here is a bit wonky

Aether700 reviewed (commented):
Much better but the name of the file will need to be updated to RuntimeSystem.java otherwise I don't think it will run.

Good improvement though.

ShayParsa committed:
Rename RunTime to RuntimeSystem.java

ShayParsa committed:
Update RuntimeSystem.java

ShayParsa committed:
Update RuntimeSystem.java

Aether700 commented on src/NAR/RuntimeSystem.java:
```diff
@@ -0,0 +1,73 @@
+public class RuntimeSystem {
+	
+	private static RuntimeSystem Instance = new RuntimeSystem();
+	RuntimeSystem() {}
```
you need to mark the constructor as private. As of now without the private keyword any other class in the same package as the RuntimeSystem class can create a RuntimeSystem object

Aether700 commented on src/NAR/RuntimeSystem.java:
```diff
@@ -0,0 +1,73 @@
+	RuntimeSystem() {}
+	
+	
+	private List listOfLines(String str) {
```
you should specify what type of object the list returned by this method is. In your case you would return a List<String>

Aether700 commented on src/NAR/RuntimeSystem.java:
```diff
@@ -0,0 +1,73 @@
+	}
+
+
+	public void runCode(String str) {
```
you forgot the static keyword for the runCode function

Aether700 commented on src/NAR/RuntimeSystem.java:
```diff
@@ -0,0 +1,73 @@
+
+		for (String line : lines) {
+
+			String statement = Instance.reduceSpace(line);
```
given that reduceSpace and isLengthGreaterThanZero are static functions you should not use the Instance variable to access them. note that java will let you do that but it's improper

Aether700 reviewed (commented):
Other than the few things I pointed out everything else is great. I tested the runtime system and it works wonderfully. Once you've cleaned up those few things we can merge it into main

ShayParsa committed:
Update RuntimeSystem.java

Aether700 committed:
adjusted the indentation of a comment

Aether700 reviewed (approved):
Great job. Now all that's left to do is officially do the tests from the documentation
